internal/testrunner/runners/system/tester.go

```diff
@@ -1460,18 +1493,42 @@ func (r *tester) validateTestScenario(ctx context.Context, result *testrunner.Re
 	if err != nil {
 		return result.WithErrorf("creating fields validator for data stream failed (path: %s): %w", r.dataStreamPath, err)
 	}
-	if errs := validateFields(scenario.docs, fieldsValidator); len(errs) > 0 {
-		return result.WithError(testrunner.ErrTestCaseFailed{
-			Reason:  fmt.Sprintf("one or more errors found in documents stored in %s data stream", scenario.dataStream),
-			Details: errs.Error(),
-		})
+
+	if r.fieldValidationMethod == allMethods || r.fieldValidationMethod == fieldsMethod {
+		if errs := validateFields(scenario.docs, fieldsValidator); len(errs) > 0 {
+			return result.WithError(testrunner.ErrTestCaseFailed{
+				Reason:  fmt.Sprintf("one or more errors found in documents stored in %s data stream", scenario.dataStream),
+				Details: errs.Error(),
+			})
+		}
 	}
 
 	err = validateIgnoredFields(r.stackVersion.Number, scenario, config)
 	if err != nil {
 		return result.WithError(err)
 	}
 
+	if r.fieldValidationMethod == allMethods || r.fieldValidationMethod == mappingsMethod {
+		logger.Warn("Validate mappings found (technical preview)")
+		mappingsValidator, err := fields.CreateValidatorForMappings(r.dataStreamPath,
+			fields.WithElasticsearchAPI(r.esAPI),
+			fields.WithIndexTemplate(scenario.indexTemplateName),
+			fields.WithDataStream(scenario.dataStream),
+			fields.WithSpecVersion(r.pkgManifest.SpecVersion),
+			fields.WithEnabledImportAllECSSChema(true),
+		)
```

Comment:
Moving the Elasticsearch API client to be mandatory in this method looks like the best move here, and I'll do that change.

I'm wondering about using an schema with both the ECS and the fields from the package.

In the current implementation, if a mapping does not exist in the preview it is checked if that path corresponds to some ECS field.
Changing the schema to be both the ECS fields and the fields from the package, could it lead to a false positive?

Thinking about an scenario where some field definition (non-ECS) in the package does not create a mapping in the preview. If the process validates with the super-set of ECS+local fields, probably it would be validated successfully that field. Could that happen? Maybe, an example of this would be fields with `type: array`.

WDYT ? @jsoriano

Reply:
Applied these changes in these two commits:
- Added "ecs" to external field: https://github.com/elastic/elastic-package/pull/2214/commits/0f980fa643b34017f31941192e7ffc0eef91cab2
- Remove LocalSchema in favor of just Schema: https://github.com/elastic/elastic-package/pull/2214/commits/e20d56397d12cc33a56a9243b6f246764ec3d60b